### ISSUE TITLE
RCS1102 - Class with no instance members that is constructed, is made static

### ISSUE THREAD

**Issue by GalaxiaGuy** [JosefPihrt/Roslynator#1103]
**Product and Version Used**: 

Roslynator.Analyzers 4.2.0

**Steps to Reproduce**:

Given a class that has no instance members but is still constructed gets changed to static.

The exact circumstances in which I found this code is a bit odd and unlikely, but it still seems like an error (since the code obviously won't compile afterwards).

**Actual Behavior**:
The following code
```csharp
    public class MainClass
    {
        public MainClass()
        {
            var weird = new WeirdClass();
        }
    }

    public class WeirdClass
    {
        public static int StaticProperty { get; }
    }
```
becomes
```csharp
    public class MainClass
    {
        public MainClass()
        {
            var weird = new WeirdClass();
        }
    }

    public static class WeirdClass
    {
        public static int StaticProperty { get; }
    }
```
**Expected Behavior**:
`WeirdClass` is not made static.

**Comment by JosefPihrt** [comment]
Hi,

Thanks for the report. This is a false positive indeed, but the problem is that the analyzer is based on analyzing the class itself, not checking if the class is instantieted anywhere in the code base (actually it's not even possible).

If this is a real code, the correct solution to this issue is to just suppress the diagnostic.
